TestingLowerBounds/Divergences/StatInfo.lean

```diff
@@ -537,39 +652,196 @@ lemma fDiv_statInfoFun_eq_integral_abs_of_nonpos_of_gt [IsFiniteMeasure μ] [IsF
+    (hβ : 0 ≤ β) (hγ : 0 ≤ γ) (hγβ : γ ≤ β) :
+    fDiv (statInfoFun β γ) μ ν = statInfo μ ν (Bool.boolMeasure (.ofReal β) (.ofReal γ))
+      + 2⁻¹ * (|β * (μ univ).toReal - γ * (ν univ).toReal|
+        + γ * (ν univ).toReal - β * (μ univ).toReal) := by
+  rw [← ENNReal.toReal_toEReal_of_ne_top statInfo_ne_top, toReal_statInfo_eq_integral_abs]
+  simp only [Bool.boolMeasure_apply_false, ENNReal.toReal_mul, hβ, ENNReal.toReal_ofReal,
+    Bool.boolMeasure_apply_true, hγ, EReal.coe_mul, EReal.coe_add, EReal.coe_neg,
+    ENNReal.toReal_toEReal_of_ne_top (measure_ne_top _ _)]
+  rw [show 2⁻¹ = ((2⁻¹ : ℝ) : EReal) from rfl, ← EReal.coe_mul_add_of_nonneg (by positivity),
+    fDiv_statInfoFun_eq_integral_abs_of_nonneg_of_le hβ hγβ]
+  simp_rw [sub_eq_add_neg, ← add_assoc]
+  rw [add_comm (-_ + _ + _)]
+  simp_rw [← add_assoc, ← sub_eq_add_neg, ]
+  rw [EReal.sub_self (EReal.coe_ne_top _) (EReal.coe_ne_bot _), zero_add]
+
+lemma fDiv_statInfoFun_eq_StatInfo_of_nonneg_of_gt [IsFiniteMeasure μ] [IsFiniteMeasure ν]
+    (hβ : 0 ≤ β) (hγ : 0 ≤ γ) (hγβ : β < γ) :
+    fDiv (statInfoFun β γ) μ ν = statInfo μ ν (Bool.boolMeasure (.ofReal β) (.ofReal γ))
+      + 2⁻¹ * (|β * (μ univ).toReal - γ * (ν univ).toReal|
+        + β * (μ univ).toReal - γ * (ν univ).toReal) := by
+  rw [← ENNReal.toReal_toEReal_of_ne_top statInfo_ne_top, toReal_statInfo_eq_integral_abs]
+  simp only [Bool.boolMeasure_apply_false, ENNReal.toReal_mul, hβ, ENNReal.toReal_ofReal,
+    Bool.boolMeasure_apply_true, hγ, EReal.coe_mul, EReal.coe_add, EReal.coe_neg,
+    ENNReal.toReal_toEReal_of_ne_top (measure_ne_top _ _)]
+  rw [show 2⁻¹ = ((2⁻¹ : ℝ) : EReal) from rfl, ← EReal.coe_mul_add_of_nonneg (by positivity),
+    fDiv_statInfoFun_eq_integral_abs_of_nonneg_of_gt hβ hγβ]
+  simp_rw [sub_eq_add_neg, ← add_assoc]
+  rw [add_comm (-_ + _ + _)]
+  simp_rw [← add_assoc, ← sub_eq_add_neg, ]
+  rw [EReal.sub_self (EReal.coe_ne_top _) (EReal.coe_ne_bot _), zero_add]
+
+-- N.B. we cannot use the Real.sign function here because it is 0 at 0, but we need it to be -1.
+lemma fDiv_statInfoFun_eq_StatInfo_of_nonneg [IsFiniteMeasure μ] [IsFiniteMeasure ν]
+    (hβ : 0 ≤ β) (hγ : 0 ≤ γ) :
+    fDiv (statInfoFun β γ) μ ν = statInfo μ ν (Bool.boolMeasure (.ofReal β) (.ofReal γ))
+      + 2⁻¹ * (|β * (μ univ).toReal - γ * (ν univ).toReal|
+        + (if γ ≤ β then -1 else 1) * (β * (μ univ).toReal - γ * (ν univ).toReal)) := by
+  rcases le_or_lt γ β with (hβγ | hβγ)
+  · rw [fDiv_statInfoFun_eq_StatInfo_of_nonneg_of_le hβ hγ hβγ, if_pos hβγ, neg_one_mul,
+      EReal.neg_sub, add_comm (-_), sub_eq_add_neg, add_assoc]
+    · exact Or.inl <| EReal.add_top_iff_ne_bot.mp rfl
+    · exact Or.inl <| Ne.symm (ne_of_beq_false rfl)
+  · rw [fDiv_statInfoFun_eq_StatInfo_of_nonneg_of_gt hβ hγ hβγ, if_neg hβγ.not_le, one_mul,
+      add_sub_assoc]
+
+lemma fDiv_statInfoFun_ne_top_of_nonneg [IsFiniteMeasure μ] [IsFiniteMeasure ν] :
+    fDiv (statInfoFun β γ) μ ν ≠ ⊤ := by
```

Comment:
```suggestion
lemma fDiv_statInfoFun_ne_top [IsFiniteMeasure μ] [IsFiniteMeasure ν] :
    fDiv (statInfoFun β γ) μ ν ≠ ⊤ := by
```
Why `of_nonneg`?

Reply:
I think that I first wrote another proof of this, involving `fDiv_statInfoFun_eq_StatInfo_of_nonneg`, that needed the hypotheses `(hβ : 0 ≤ β) (hγ : 0 ≤ γ)`, then I realized that there was a much shorter proof without those hypotheses, but I forgot to change the name accordingly.